src/sql/base/browser/ui/infoButton/infoButton.ts

```diff
@@ -0,0 +1,171 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'vs/css!./infoButton';
+import { Button as sqlButton } from 'sql/base/browser/ui/button/button';
+import * as DOM from 'vs/base/browser/dom';
+import { IButtonOptions } from 'vs/base/browser/ui/button/button';
+
+
+// type IUserFriendlyIcon = string | URI | { light: string | URI; dark: string | URI };
+
+export interface IInfoButtonOptions extends IButtonOptions {
+	buttonMaxHeight: number,
+	buttonMaxWidth: number,
+	description: string,
+	iconClass: string,
+	iconHeight: number,
+	iconWidth: number,
+	title: string,
+}
+
+export class InfoButton extends sqlButton {
+	private _container: HTMLElement;
+	private _main: HTMLElement;
+	private _iconContainer: HTMLElement;
+	private _iconElement: HTMLElement;
+	private _textContainer: HTMLElement;
+	private _pTitle: HTMLElement;
+	private _pDesc: HTMLElement;
+
+	private _buttonMaxHeight?: number;
+	private _buttonMaxWidth?: number;
+	private _description?: string;
+	private _iconClass?: string;
+	private _iconHeight?: number;
+	private _iconWidth?: number;
+	private _title?: string;
+
+	constructor(container: HTMLElement, options?: IInfoButtonOptions) {
+		super(container, options);
+		this._container = container;
+
+		this._container.style.display = 'flex';
+		this._container.style.justifyContent = 'space-around';
+
+		this._main = document.createElement('div');
+		DOM.addClass(this._main, 'flexContainer');
+		this._main.style.cursor = 'pointer';
+		this._main.style.padding = '10px';
+
+		this._iconContainer = document.createElement('div');
```

Comment:
any reason these can't be css files?

Reply:
No. It was only recently that I had added a stylesheet for the button. Would you like all properties moved into CSS instead?

Reply:
I think that makes more sense